Documentation/user-guides/scrapeconfig.md

```diff
@@ -0,0 +1,162 @@
+* `http_sd`
+* `kubernetes_sd`
+
+Also, the following examples are basic and the CRD is constantly evolving, adding new features. Check the API documentation to see all the available fields.
+
+If you have an interest in another service discovery mechanism or you see something missing in the implementation, please
+[open an issue](https://github.com/prometheus-operator/prometheus-operator/issues).
+
+## `static_config`
+
+For example, to scrape a target located at `http://10.1.2.3:9100`, use the following:
+
+```yaml
+apiVersion: coreos.monitoring.com/v1alpha1
+kind: ScrapeConfig
+metadata:
+  name: my-scrape-config
+  namespace: my-namespace
+spec:
+  staticConfigs:
+    - labels:
+        job: mytarget
+      targets: 
+      - 10.1.2.3:9100
+    - labels:
+        job: prometheus
+      targets:
+        - prometheus.demo.do.prometheus.io:9090
+```
+
+## `file_sd`
+
+To use `file_sd`, a file has to be mounted in the Prometheus or PrometheusAgent pods. The following configmap is a service discovery file:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: scrape-file-sd-targets
+  namespace: monitoring
+data:
+  targets.yaml: |
+    - labels:
+        job: node-demo
+      targets:
+      - node.demo.do.prometheus.io:9100
+    - labels:
+        job: prometheus
+      targets:
+      - prometheus.demo.do.prometheus.io:9090
+```
+
+This `ConfigMap` will then need to be mounted in the `Prometheus` spec:
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+metadata:
+  name: your-prometheus
+  namespace: monitoring
+spec:
+  scrapeConfigSelector:
+    prometheus: system-monitoring-prometheus
+  configMaps:
+    - scrape-file-sd-targets
+```
+
+You can then use ScrapeConfig to reference that file and scrape the associated targets:
+
+```yaml
+apiVersion: monitoring.coreos.com/v1alpha1
+kind: ScrapeConfig
+metadata:
+  name: scrape-config-example
+  namespace: my-namespace
+  labels:
+    app.kubernetes.io/name: scrape-config-example
+spec:
+  fileSDConfigs:
+    - files:
+        - /etc/prometheus/configmaps/scrape-file-sd-targets/targets.yaml
+      refreshInterval: 15s
+```
+
+## `http_sd`
+
+`http_sd` works the same way as the `file_sd` but requires an endpoint providing that data rather than a file. For instance:
+
+```yaml
+apiVersion: monitoring.coreos.com/v1alpha1
+kind: ScrapeConfig
+metadata:
+  name: scrape-config-example
+  namespace: my-namespace
+  labels:
+    app.kubernetes.io/name: scrape-config-example
+spec:
+  httpSDConfigs:
+    - url: http://my-external-api/discovery
+      refreshInterval: 15s
+```
+
+## `kubernetes_sd`
+
+`kubernetes_sd` allows retrieving scrape targets from Kubernetes' REST API. Here is an example that selects nodes in the kubernetes cluster:
+
+```yaml
+apiVersion: monitoring.coreos.com/v1alpha1
+kind: ScrapeConfig
+metadata:
+  name: scrape-config-example
+  namespace: my-namespace
+  labels:
+    app.kubernetes.io/name: scrape-config-example
+spec:
+  kubernetesSDConfig:
```

Comment:
We would need a TLSConfig and Authorization section SC spec 
here is the example I tested https://github.com/prometheus-operator/prometheus-operator/pull/5675#issuecomment-1608867483

Also cluster role binding was required for prometheus so that prometheus serviceAccount has access to list Node objects otherwise prometheus errors about access issue so we should mention that as well

Reply:
Following I will remove `kubernetes_sd` to keep it short and we can figure out a specific documentation for it as there is more context than that (e.g. "How to scrape an exporter located on all the kubernetes worker").